push_gateway_pruner.py

```diff
@@ -0,0 +1,212 @@
+
+    Must start with job key/value. Order of remaining keys doesn't matter
+    and will match any group with the same set of labels.
+
+    There is a bug where if the group is /job/<job>/instance/<EMPTY STRING> and there are no other labels,
+    it wouldn't be able to delete it for some reason. Extreme edge case since instance should almost always be provided.
+    """
+
+    if "job" not in labels:
+        raise ValueError(f"Job key not found in labels. Labels: {labels}")
+
+    url = urllib.parse.urljoin(pushgateway_url, f"metrics/job/{labels['job']}")
+    for key, value in labels.items():
+        if key != "job":
+            # empty strings need to be replaced with '='
+            urlsafe_value = (
+                base64.urlsafe_b64encode(bytes(value, "utf-8")).decode("utf-8")
+                if len(value) > 0
+                else "="
+            )
+            url += f"/{key}@base64/{urlsafe_value}"
+    return url
+
+
+@click.command()
+@click.option(
+    "--prune_interval_seconds",
+    default=600,
+    type=int,
+    help="How often to start the pruning job in seconds. Default is 600.",
+)
+@click.option(
+    "--pushgateway_url",
+    default="http://localhost:9091/",
+    type=str,
+    help="URL for your pushgateway instance (without /metrics postfix). Default is http://localhost:9091/",
+)
+@click.option(
+    "--prune_threshold_seconds",
+    default=600,
+    type=int,
+    help="Number of seconds since <prune_metric_name>'s value after which to prune a sample. Default is 600.",
+)
+@click.option(
+    "--requests_timeout",
+    default=10,
+    type=int,
+    help="Timeout in seconds for the Get request to <pushgateway url>/metrics. Default is 10.",
+)
+@click.option(
+    "--prune_metric_name",
+    default="push_time_seconds",
+    type=str,
+    help="The name of the metric which is used as each group's start time. Default is 'push_time_seconds'.",
+)
+@click.option(
+    "--logging_level",
+    default="INFO",
+    type=str,
+    help="Logging Level",
+)
+def main(
+    prune_interval_seconds,
+    pushgateway_url,
+    prune_threshold_seconds,
+    requests_timeout,
+    prune_metric_name,
+    logging_level,
+):
+    try:
+        logger.setLevel(logging_level)
+
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(logging.DEBUG)
+        formatter = logging.Formatter(
+            "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+        )
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
+        start_interval(
+            prune_interval_seconds,
+            pushgateway_url,
+            prune_threshold_seconds,
+            requests_timeout,
+            prune_metric_name,
+        )
+
+    except Exception as e:
+        logger.error(
+            f"Error for push-gateway-pruner pruning pushgateway at {pushgateway_url}", e
+        )
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
Change to `sys.exit(main())` and return 0 in your main function. 
https://docs.python.org/3/library/__main__.html
So that exit codes are propagated.